README.md

```diff
@@ -157,6 +160,24 @@ Echo.join('channel3');
 Echo.leaveChannel("presence-channel3")
 ```
 
+**2. Error handling**
+- [Ably echo client](https://github.com/ably-forks/laravel-echo) emit [ably specific error codes](https://github.com/ably/ably-common/blob/main/protocol/errors.json) instead of [pusher error codes](https://pusher.com/docs/channels/library_auth_reference/pusher-websockets-protocol/#error-codes).
+- Aim is to make sure error details are descriptive and easy to understand, so it's more effective to take a corrective action ( Pusher errors lack error context, mostly emitted as integer error codes ).
+- Ably errors are provided as a [errorInfo object](https://ably.com/docs/api/realtime-sdk/types#error-info) with proper error context.
+- Care needs to be taken while checking on the errorInfo object and mapping needs to be done from [pusher error codes](https://pusher.com/docs/channels/library_auth_reference/pusher-websockets-protocol/#error-codes) to [ably error codes](https://github.com/ably/ably-common/blob/main/protocol/errors.json).
+
+```js
+    channel.error(error => {
+        if (error && error.code === 40142) { // ably token expired
+            console.error(error);
+            // take corrective action on UI
+        }
+    })
+```
+**Note :**
+- AblyPresenceChannel -> `here` method gets called everytime client **joins, updates or leaves** the channel unlike pusher client (gets called only for the first time).
+- This is perfectly in sync with interface method [**here** provided under **presence-channel.ts**](https://github.com/laravel/echo/blob/master/src/channel/presence-channel.ts#L10).
```

Comment:
I'm not entirely sure what this sentence means, could you clarify for me pls?

Reply:
`here` method for pusher gets called only for the first time. This is supposed to be called whenever user joins, updates or leaves the presence as per echo interface.

Reply:
yeah i get line 178, i'm asking about line 179 though

Reply:
It means that it follows the implementation as per the standard PresenceChannel interface.

Reply:
Also, means `pusher` is doing it wrong : )

Reply:
Oh I see, that makes sense, I think you should remove this line then, it's quite confusing and doesn't seem necessary

Reply:
I have remove the line, and updated it with line that will actually make sense.